src/main/java/com/lambdaworks/redis/protocol/ConnectionWatchdog.java

```diff
@@ -63,6 +64,7 @@
     private long lastReconnectionLogging = -1;
     private CommandHandler<?, ?> commandHandler;
 
+    private final AtomicBoolean isWaitingReconnection;
     private volatile int attempts;
     private volatile boolean listenOnChannelInactive;
     private volatile Timeout reconnectScheduleTimeout;
```

Comment:
Not sure the benefit of keeping this field, I think this make setting back to null a little hard.

Reply:
https://github.com/lettuce-io/lettuce-core/pull/679/files#diff-e9f8850c0db9fa05de5c3c9d18f1269eR237